HW03/matsvei_basiayeu/matsvei_basiayeu_hw_03_t_02.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# thi is class homework2
+class Homework3
+  SCAN_REGEX = %r{\d+\.\d\.\d+\.\d+\s-\s-\s\[\d+/\w+/\d+:\d+:\d+:\d+\s\+\d+\]\s"\w+\s/\S+\s\w+\S+\s\d+\s\S+\s\S+}
+  SPLIT_REGEX = /[\[(.*)\]]/
+
+  def task2(log)
+    log.split("\n").delete_if { |string| good_format?(string) != string }.map { |str| print_string(str) }
+  end
+
+  private
+
+  def good_format?(string)
+    string.scan(SCAN_REGEX).join
+  end
+
+  def print_string(string)
+    "DATE: #{string.split(SPLIT_REGEX)[4]} FROM: #{string.split(' ')[0]} TO: #{string.split(' ')[6].upcase}"
```

Comment:
I would write a regular expression for address, ip and time and use them to select data from selected rows
up to u

Reply:
I fixed it. Is that right?

REGEX_DATE = %r{\d+/\w+/\d+\:\d+\:\d+\:\d+\s\+\d+}
  REGEX_IP = /\d+\.\d+\.\d+\.\d+/
  REGEX_TEXT = %r{\w+/\d/\w+\s\w+/\d\.\d}


"DATE: #{string.match(REGEX_DATE)} FROM: #{string.match(REGEX_IP)} TO: #{string.match(REGEX_TEXT).to_s.upcase}"